packages/backend-common/src/database/SingleDatabase.ts

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Knex from 'knex';
+import { Config } from '@backstage/config';
+import { createDatabaseClient, ensureDatabaseExists } from './connection';
+import {
+  DatabaseConfiguration,
+  PluginDatabaseFactory,
+  PluginDatabaseManager,
+} from './types';
+
+export class SingleDatabaseConfiguration implements DatabaseConfiguration {
+  private readonly config: Config;
+
+  constructor(config: Config) {
+    this.config = config;
+  }
+
+  getDatabaseConfig(_pluginId: string): Config {
+    return this.config;
+  }
+
+  getAdminConfig(): Config {
+    return this.config;
+  }
+}
+
+export class SingleDatabaseManager implements PluginDatabaseManager {
+  private readonly config: DatabaseConfiguration;
+
+  constructor(config: DatabaseConfiguration) {
+    this.config = config;
+  }
+
+  getDatabaseFactory(pluginId: string): PluginDatabaseFactory {
```

Comment:
I would maybe pass an `options` argument here rather then ordinal to allow for future args. For example:

```ts
getDatabaseFactory({ pluginId }: { pluginId: string }): PluginDatabaseFactory {
}
```

Reply:
Will this likely need more args though?

Reply:
🤔 Actually maybe not, now that I think of it.

Reply:
@andrewthauer suggested that we hand an object to the plugin, that way we can extend the interface if needed in future. Implemented in https://github.com/spotify/backstage/pull/2697/commits/35c0294018b8433787a795b7d9fcb6e9daef2b68.